dnf-behave-tests/createrepo_c/steps/repodata.py

```diff
@@ -32,11 +33,46 @@
       "md_ns": "http://linux.duke.edu/metadata/repo"}
 
 
+def keys_do_not_differ(prim, flist, oth):
+    if set(prim.keys()) != set(flist.keys()):
+        print_lines_diff(prim.keys(), flist.keys())
```

Comment:
You're converting the `prim.keys()` to a set for the check itself. I'm not sure if it's "just in case" or actually needed (i.e. there's either a possibility of duplicate values or perhaps the order can differ?), but in either case the diff that is printed will be confusing, because either there will be duplicities or the order will be wrong.

I wouldn't mind doing `print_lines_diff(typle(set(prim.keys())), tuple(set(flist.keys())))` as it doesn't matter much here (though I'm not entirely sure the order will be stable this way), but I think you can just as easily extend `print_lines_diff()` to support sets?

Reply:
I just removed the set conversion, I don't remember why I put it there since duplicates and or different order would actually indicate a problem. Thanks. :-)